api/setting.go

```diff
@@ -68,3 +76,16 @@ func (find *SettingFind) String() string {
 	}
 	return string(str)
 }
+
+// SettingAppIMValue is the setting value of SettingAppIM type setting.
+type SettingAppIMValue struct {
+	IMType           string `json:"imType"`
+	AppID            string `json:"appId"`
+	AppSecret        string `json:"appSecret"`
+	ExternalApproval struct {
+		Enabled bool `json:"enabled"`
+		Payload struct {
+			ApprovalCode string `json:"approvalCode"`
```

Comment:
Per chat, let's move ApprovalCode under ExternalApproval instead of introducing a payload.

Reply:
Updated